### PR TITLE
Add contribute link to navbar

### LINKED ISSUE #7237: Add "Contributing" to the navigation bar?
### Enter your suggestions in details:

This came from a Bluesky thread and somehow we went a bit off-topic to discuss about how to find the useful information about contributing to the project https://bsky.app/profile/patak.dev/post/3lbash7plsk2q

Currently, there is a page on the website about contributing but it's a bit hidden: in About -> Contribute to Node.js: https://nodejs.org/en/about/get-involved/contribute

Considering the project has always been very understaffed, I think it would be overall better for the project if we can have a more prominent place for the information about contribution on the website, say putting it in the navigation bar.

### PR DESCRIPTION
**Pull request by bmuenzenmeyer**
## Description

Does the simplest thing we can to improve visibility of contribution, while avoiding some of the problems we encountered with the version support docs.

As @joyeecheung accurately points out, directly linking to this content is the best way to keep it up to date.

Open to iteration, but I wanted to advance the conversation with a concrete implementation.

## Validation

See the preview build

image

## Related Issues

closes #7237

### Check List

- [x] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [x] I have run `npm run format` to ensure the code follows the style guide.
- [x] I have run `npm run test` to check if all tests are passing.
- [x] I have run `npx turbo build` to check if the website builds without errors.
- [x] I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

**Review by mikeesto — APPROVED** [review]
LGTM, I think this is a good approach given the issues discussed

**Review by AugustinMauroy — APPROVED** [review]

File: apps/site/navigation.json [hunk 1]
```diff
@@ -20,6 +20,11 @@
       "link": "https://nodejs.org/docs/latest/api/",
       "label": "components.containers.navBar.links.docs"
     },
+    "contribute": {
```
Comment (ovflowd) [inline apps/site/navigation.json:23]: OOC -- what is the point of a contribute page within the website if we simply redirect here? Should we then simply delete the page and make the link on the sidebar also redirect here? I'm fine with that TBH and possibly is the best thing to be done here, can you do that?

Comment (bmuenzenmeyer) [reply apps/site/navigation.json:23]: I can certainly do that, but as we've learned recently, this is a [CONTENT VS CODE decision](https://github.com/nodejs/nodejs.org/blob/main/CONTENT_VS_CODE.md).

I've audited the content on https://nodejs.org/en/about/get-involved/contribute, comparing it to https://github.com/nodejs/node/blob/main/CONTRIBUTING.md. I want to make sure we are not net-losing content that may not be present within https://github.com/nodejs/node/blob/main/CONTRIBUTING.md


| website section | comparative location on main repo | plan |
| -- | -- | -- |
| [asking for general help](https://nodejs.org/en/about/get-involved/contribute#asking-for-general-help) | [linked on the doc](https://github.com/nodejs/node/blob/main/CONTRIBUTING.md#issues) | ✅ already handled |
| [reportng an issue, general](https://nodejs.org/en/about/get-involved/contribute#reporting-an-issue) | [linked on the doc](https://github.com/nodejs/node/blob/main/CONTRIBUTING.md#issues) | ✅ already handled |
| reporting an issue specific to the website | nothing | do nothing, expect to triage and move issues? |
 | reporting and issue, [website status](https://status.nodejs.org/) link | nothing | non-blocking in my opinion,  [already on the README of nodejs.org](https://github.com/nodejs/nodejs.org). Add to [org-level README](https://github.com/nodejs/.github/blob/main/profile/README.md) or [nodejs/node README](https://github.com/nodejs/node) as other work  |
| [Code contributions](https://nodejs.org/en/about/get-involved/contribute#code-contributions) | [linked on the doc](https://github.com/nodejs/node/blob/main/doc/contributing/pull-requests.md) | ✅ already handled|
| [node todo link](https://www.nodetodo.org/) | [differing content](https://github.com/nodejs/node/blob/main/doc/contributing/pull-requests.md#setting-up-your-local-environment) | ✅ already handled |
| [becoming a collaborator](https://nodejs.org/en/about/get-involved/contribute#becoming-a-collaborator) | [linked to, found within GOVERNANCE.md](https://github.com/nodejs/node/blob/main/CONTRIBUTING.md#contributing-to-nodejs) | ✅ already handled|

Comment (ovflowd) [reply apps/site/navigation.json:23]: Technically speaking we wouldn't be authoring content; But removing it, which IMO we've done plenty of times before 👀

**Comment by bmuenzenmeyer** [comment]
todo: we also need a redirect

**Review by aymen94 — APPROVED** [review]

File: apps/site/navigation.json (hunk 1 above)
Comment (bmuenzenmeyer) [reply apps/site/navigation.json:23]: resolved via https://github.com/nodejs/nodejs.org/pull/7274/commits/4093ba29370def1c77c8e92e2a763df154a7d76c

**Comment by bmuenzenmeyer** [comment]
@joyeecheung this was your original idea. we now have:

- top-level contributing link going to the nodejs/node CONTRIBUTING file
- contributing link in the about/get involved going to the same
- a redirect in place to the same
- deleted the duplicate content

can you think of anything else?

**Comment by joyeecheung** [comment]
Thanks for opening the PR! Looks good as far as I can tell, but I do wonder if we should keep the original contribute page and just leave a link in there to CONTRIBUTING.md, in case there are any external websites already linking to the removed page and the URL would be broken (if we think nobody is linking to it then it's fine to just remove, though it seems plausible to me).

**Comment by mikeesto** [comment]
> Thanks for opening the PR! Looks good as far as I can tell, but I do wonder if we should keep the original contribute page and just leave a link in there to CONTRIBUTING.md, in case there are any external websites already linking to the removed page and the URL would be broken (if we think nobody is linking to it then it's fine to just remove, though it seems plausible to me).

That's a good point although I think this redirect should help avoid breaking things: https://github.com/nodejs/nodejs.org/pull/7274/files#diff-5ca5a80a6f81808f1779e8216ad41ca807d50397070d24bc8b4cc72653c84f0bR268-R269

**Review by AugustinMauroy — APPROVED** [review]

**Comment by joyeecheung** [comment]
Oh right I didn't understand what it was doing at first. In that case LGTM, thanks!

**Pull request merged**